modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java

```diff
@@ -921,6 +923,100 @@ public void testAddOrUpdateCustomObjectField() throws Exception {
 			_objectDefinitionLocalService, new String[] {"C_A", "C_AA"});
 	}
 
+	@Test
+	public void testAddOrUpdateSystemObjectField() throws Exception {
+		ObjectDefinition modifiableSystemObjectDefinition =
+			ObjectDefinitionTestUtil.addModifiableSystemObjectDefinition(
+				TestPropsValues.getUserId(), null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"Test", null, null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				ObjectDefinitionConstants.SCOPE_SITE, null, 1,
+				_objectDefinitionLocalService, Collections.emptyList());
+
+		ObjectField objectField =
+			_objectFieldLocalService.addOrUpdateSystemObjectField(
+				null, TestPropsValues.getUserId(), 0L,
+				modifiableSystemObjectDefinition.getObjectDefinitionId(),
+				ObjectFieldConstants.BUSINESS_TYPE_TEXT, null, null,
+				ObjectFieldConstants.DB_TYPE_STRING, false, true, "",
+				LocalizedMapUtil.getLocalizedMap("Able"), "able",
+				ObjectFieldConstants.READ_ONLY_FALSE, null, false, false,
+				Collections.emptyList());
+
+		Assert.assertEquals("able", objectField.getDBColumnName());
+		Assert.assertEquals(
+			ObjectFieldConstants.DB_TYPE_STRING, objectField.getDBType());
+		Assert.assertFalse(objectField.isIndexed());
+		Assert.assertTrue(objectField.isIndexedAsKeyword());
+		Assert.assertEquals("", objectField.getIndexedLanguageId());
+		Assert.assertEquals(
+			LocalizedMapUtil.getLocalizedMap("Able"),
+			objectField.getLabelMap());
+		Assert.assertEquals("able", objectField.getName());
+		Assert.assertFalse(objectField.isRequired());
+		Assert.assertTrue(objectField.isSystem());
+
+		String liferayMode = SystemProperties.get("liferay.mode");
+
+		SystemProperties.clear("liferay.mode");
+
+		try {
+			objectField = _objectFieldLocalService.addOrUpdateSystemObjectField(
+				objectField.getExternalReferenceCode(),
+				TestPropsValues.getUserId(), 0L,
+				modifiableSystemObjectDefinition.getObjectDefinitionId(),
+				ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+				RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+				ObjectFieldConstants.DB_TYPE_STRING, false, true, "",
+				LocalizedMapUtil.getLocalizedMap("Charlie"), "able",
+				ObjectFieldConstants.READ_ONLY_FALSE, null, true, false,
+				Collections.emptyList());
+
+			Assert.assertEquals("able", objectField.getDBColumnName());
+			Assert.assertEquals(
+				ObjectFieldConstants.DB_TYPE_STRING, objectField.getDBType());
+			Assert.assertFalse(objectField.isIndexed());
+			Assert.assertTrue(objectField.isIndexedAsKeyword());
+			Assert.assertEquals("", objectField.getIndexedLanguageId());
+			Assert.assertEquals(
+				LocalizedMapUtil.getLocalizedMap("Charlie"),
+				objectField.getLabelMap());
+			Assert.assertEquals("able", objectField.getName());
+			Assert.assertFalse(objectField.isRequired());
+			Assert.assertTrue(objectField.isSystem());
+		}
+		finally {
+			SystemProperties.set("liferay.mode", liferayMode);
```

Comment:
I didn't understand what we need to  repeat the same test.

Reply:
It's not the same case here. The goals are:

- Test if it's possible to add system object fields with the `addOrUpdateSystemObjectField` method.
- Test if it's not possible create system object fields in not allowed bundles
- Test if it's possible to create system object fields in allowed bundles